lib/Controller/LoginController.php

```diff
@@ -356,8 +357,20 @@ public function code($state = '', $code = '', $scope = '') {
 		$this->logger->debug('Logging user in');
 
 		$this->userSession->setUser($user);
-		$this->userSession->completeLogin($user, ['loginName' => $user->getUID(), 'password' => '']);
+		
+		$this->userSession->getSession()->regenerateId();
+		$tokenProvider = \OC::$server->query(DefaultTokenProvider::class);
+		$this->userSession->setTokenProvider($tokenProvider);
```

Comment:
Is this really necessary, why would it need to use enforce the DefaultTokenProvider? 

Do you have a more detailed list of steps to reproduce the issue? Maybe you can also share your Nextcloud config in place to check for the session timeout variables, as I so far didn't manage to reproduce any issue.